### DISCUSSION TITLE
Another potential idea

### DISCUSSION THREAD

**Discussion by Udbhavkush** [Udbhavkush/NLP_Final_Project_Team_4#3] [Ideas]
Translate and summarize news
You can build a web app that translates news from Arabic to English and summarizes them, using great Python libraries like [newspaper](https://github.com/codelucas/newspaper),[ transformers](https://github.com/huggingface/transformers), and[ gradio](https://gradio.app/).

Where:

[Newspaper3k (11.1k stars)](https://github.com/codelucas/newspaper): scrape almost any news website
[HuggingFace Transformers (48k)](https://github.com/huggingface/transformers): use state-of-the-art natural language models
[Gradio (2.9k)](https://gradio.app/): build interactive web-based demos
Start Project Now → [Useful Link](https://abidlabs.github.io/Summarize-News/)

Basically we can scrape any website from the web using the GitHub link above. Then we can translate the news and summarize it.
So, we scrape any news website that posts news in say Hindi. We will be able to showcase multiple concepts like scraping, translation and summarizer.

**Comment by ianjali** [comment]
1. Transformer-Based Code Commenting :  [https://arxiv.org/pdf/2107.03644.pdf](url)
2. multilingual model : [https://oneai.com/learn/multilingual-nlp](url)

**Comment by sunishaharish** [comment]
Automated Fact checking: [https://direct.mit.edu/tacl/article/doi/10.1162/tacl_a_00454/109469/A-Survey-on-Automated-Fact-Checking]

**Comment by ianjali** [comment]
------------------
minutes of meeting 
------------------
-> mental health 
	- multiple database, distressing messages 
	- problem : so much work has been done 
	- falls under sentiment analysis
	- classification
	- text generation - chat bot - try taking these preventive steps
	- can make it multilingual as well  
	- language translation - replies could be in that native language and also in english 
	- https://github.com/kharrigian/mental-health-datasets
	
	
-> transformer based code commenting ()
	- unique idea 
	- might struggle a little bit
	- https://github.com/github/CodeSearchNet
	
based on the above discussion, we should focus on mental health since we get to work with problems like text generation, classification/sentiment analysis, language translation.
